File: python/backend/src/app/models/rag.py
"""Pydantic models for RAG API."""

from __future__ import annotations

from pydantic import BaseModel, Field


class RagQueryRequest(BaseModel):
    query: str
    top_k: int = Field(default=5, ge=1, le=20)


class RagChunk(BaseModel):
    note_path: str
    content: str
    score: float


class RagQueryResponse(BaseModel):
    answer: str
    sources: list[RagChunk]

File: packages/shared-types/src/rag.ts
export interface IRagQueryRequest {
  query: string;
  top_k?: number;
}

export interface IRagChunk {
  note_path: string;
  content: string;
  score: number;
}

export interface IRagSource {
  note_path: string;
  content: string;
  score: number;
}

export interface IRagQueryResponse {
  answer: string;
  sources: IRagChunk[];
}
